src/Models/Services/MsalAuthService.ts

```diff
@@ -1,81 +1,189 @@
+                )
+                    .then((tenants: any) => {
+                        let resolvedTenants;
+                        try {
+                            const responsePayload = JSON.parse(tenants);
+                            resolvedTenants = responsePayload.value;
+                        } catch (err) {
+                            resolvedTenants = [];
+                            console.log(err);
+                        }
+                        continuation?.({
+                            name,
+                            userName,
+                            tenants: resolvedTenants,
+                            accounts
+                        });
+                    })
+                    .catch((err) => {
+                        console.log(err);
+                        continuation?.({
+                            name,
+                            userName,
+                            tenants: [],
+                            accounts
+                        });
+                    });
+            });
+        };
+
+        this.MSALObj.handleRedirectPromise().then(
+            (resp: AuthenticationResult | null) => {
+                if (resp !== null) {
                     this.isLoggingIn = false;
-                    this.shiftAndExecuteGetTokenCall();
-                })
-                .catch(function (error) {
-                    //login failure
-                    alert(error);
-                });
-        }
+                    const activeAccount = resp.account;
+                    this.setActiveAccount(activeAccount);
+                    if (continuation) {
+                        getTenantsAndContinuation(
+                            activeAccount.name,
+                            activeAccount.username,
+                            [activeAccount]
+                        );
+                    } else {
+                        this.ssoSilent(activeAccount.username);
+                    }
+                    for (let i = 0; i < this.getTokenCalls.length; i++) {
+                        this.shiftAndExecuteGetTokenCall();
+                    }
+                } else {
+                    this.isLoggingIn = false;
+                    const allAccounts = this.MSALObj.getAllAccounts();
+                    if (!allAccounts || allAccounts.length < 1) {
+                        this.MSALObj.loginRedirect(this.loginRedirectRequest);
+                    } else {
+                        let activeAccount;
+                        if (this.tenantId === DEFAULT_TENANT_ID) {
+                            if (
+                                this.targetTenantId &&
+                                this.targetTenantId !== DEFAULT_TENANT_ID
+                            ) {
+                                activeAccount = allAccounts.find(
+                                    (a: any) =>
+                                        a.tenantId === this.targetTenantId
```

Comment:
I added target Id to capture intended tenant id passed in at the time of construction (e.g. if there is tenant id from url or local storage etc.)